### PR TITLE
23 cart model

### PR DESCRIPTION
**Pull request by mhrownaghi**
Create cart and cart item model without test methods

### CODE REVIEW

File: database/migrations/2023_02_19_133321_create_carts_table.php [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('carts', function (Blueprint $table) {
+            $table->id();
+            $table->string('customer_email');
+            $table->string('customer_firstname');
+            $table->string('customer_lastname');
+            $table->foreignId('customer_id');
```
Comment (AmooAti) [inline database/migrations/2023_02_19_133321_create_carts_table.php:21]: refrence and nullOnDelete

File: app/Models/Cart.php [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+use Illuminate\Database\Eloquent\Model;
+
+class Cart extends Model
+{
+    use HasFactory;
+
+    protected $fillable = [
+        'customer_email',
+        'customer_firstname',
+        'customer_lastname',
```
Comment (AmooAti) [inline app/Models/Cart.php:15]: customer_id?

File: database/migrations/2023_02_19_143928_create_cart_items_table.php [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+     */
+    public function up()
+    {
+        Schema::create('cart_items', function (Blueprint $table) {
+            $table->id();
+            $table->unsignedInteger('qty');
+            $table->string('name');
+            $table->unsignedInteger('price');
+            $table->foreignId('cart_id');
+            $table->foreignId('product_id');
```
Comment (AmooAti) [inline database/migrations/2023_02_19_143928_create_cart_items_table.php:22]: refrence and nullOnDelete?

File: database/migrations/2023_02_19_143928_create_cart_items_table.php [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('cart_items', function (Blueprint $table) {
+            $table->id();
+            $table->unsignedInteger('qty');
+            $table->string('name');
+            $table->unsignedInteger('price');
+            $table->foreignId('cart_id');
```
Comment (AmooAti) [inline database/migrations/2023_02_19_143928_create_cart_items_table.php:21]: refrence and cascade on delete?

File: app/Models/CartItem.php [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+{
+    use HasFactory;
+
+    protected $fillable = [
+        'qty',
+        'name',
+        'price',
+    ];
+
+}
```
Comment (AmooAti) [inline app/Models/CartItem.php:18]: Cart relation?

File: app/Models/Cart.php [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+        'shipping_method',
+        'items_count',
+        'item_qty',
+        'grand_total',
+        'subtotal',
+        'shipping_cost',
+        'discount_amount',
+        'is_active'
+    ];
+}
```
Comment (AmooAti) [inline app/Models/Cart.php:25]: items relation?

File: database/migrations/2023_02_19_133321_create_carts_table.php (hunk 1 above)
Comment (mhrownaghi) [reply database/migrations/2023_02_19_133321_create_carts_table.php:21]: I resolve it with this code:
`            $table->foreignId('customer_id')
                ->constrained()
                ->nullOnDelete();
`

File: database/migrations/2023_02_19_143928_create_cart_items_table.php (hunk 3 above)
Comment (mhrownaghi) [reply database/migrations/2023_02_19_143928_create_cart_items_table.php:22]: If a product removed then cart's item that related to this product must be removed too. it's true?
too
if a cart removed then cart's items that related to this cart must be removed too. it's true?

**Review by AmooAti — APPROVED** [review]

**Pull request merged**